src/quo/components/wallet/token_input/view.cljs

```diff
@@ -35,85 +20,63 @@
-           :icon-only?          true
-           :size                32
-           :on-press            handle-on-swap
-           :type                :outline
-           :accessibility-label :reorder}
-          :i/reorder]]))))
+  [{:keys [token-symbol on-token-press value error? on-swap currency-symbol]}]
+  (let [theme        (quo.theme/use-theme)
+        window-width (:width (rn/get-window))]
+    [rn/pressable
+     {:style    {:width          "100%"
+                 :flex-direction :row}
+      :on-press on-token-press}
+     [token/view
+      {:token token-symbol
+       :size  :size-32}]
+     [rn/view {:style (style/input-container window-width)}
+      [rn/text-input
+       {:style                  (style/text-input theme error?)
+        :placeholder-text-color (style/placeholder-text theme)
+        :placeholder            "0"
+        :keyboard-type          :numeric
+        :editable               false
+        :value                  value}]
+      [token-name-text theme currency-symbol]]
+     [button/button
+      {:icon                true
+       :icon-only?          true
+       :size                32
+       :on-press            #(when on-swap (on-swap))
+       :type                :outline
+       :accessibility-label :reorder}
+      :i/reorder]]))
 
 (defn- view-internal
-  [{:keys [container-style on-swap crypto?] :as props}]
-  (let [theme                               (quo.theme/use-theme)
-        width                               (:width (rn/get-window))
-        [value-internal set-value-internal] (rn/use-state nil)
-        handle-on-swap                      (rn/use-callback
-                                             (fn []
-                                               (when on-swap (on-swap (not crypto?))))
-                                             [on-swap])]
+  [{:keys [token-symbol
+           value
+           on-token-press
+           error?
+           container-style
+           on-swap
+           converted-value
+           hint-component
+           currency-symbol]}]
+  (let [theme (quo.theme/use-theme)
+        width (:width (rn/get-window))]
     [rn/view {:style (merge (style/main-container width) container-style)}
      [rn/view {:style style/amount-container}
       [input-section
-       (assoc props
-              :value-internal     value-internal
-              :theme              theme
-              :set-value-internal set-value-internal
-              :handle-on-swap     handle-on-swap
-              :crypto?            crypto?)]]
+       {:theme           theme
+        :token-symbol    token-symbol
+        :on-token-press  on-token-press
+        :value           value
+        :error?          error?
+        :on-swap         on-swap
+        :currency-symbol currency-symbol}]]
      [divider-line/view {:container-style (style/divider theme)}]
-     [data-info (assoc props :theme theme)]]))
+     [rn/view {:style style/data-container}
+      hint-component
```

Comment:
Component in the left bottom corner moved out because new screens will use another one in similar token input